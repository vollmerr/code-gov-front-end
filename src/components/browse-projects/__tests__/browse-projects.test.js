import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import BrowseProjects from 'components/browse-projects';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Testing <BrowseProjects /> component', () => {
  const props = {
    onFilterBoxChange: jest.fn(),
    onFilterTagClick: jest.fn(),
    onSortChange: jest.fn(),
    saveFilterData: jest.fn(),
    updatePage:  jest.fn(),
    browseParams: {
      filters: [],
      page: 1,
      size: 10,
      sort: "data_quality"
    },
    browseResults: {
      params: {
        page: 1,
        size: 10,
        sort: "data_quality",
        agencies: [],
        languages: [],
        licenses: [],
        usageTypes: []
      },
      repos: [{
        repoID: 'id-1',
      }]
    },
    filters: {
      agencies: [{name: 'test-name', value: 'TV'}],
      languages: [{name: 'test-name', value: 'TV'}],
      licenses: [{name: 'test-name', value: 'TV'}],
      usageTypes: [{name: 'test-name', value: 'TV'}]
    }
  };
  const store = mockStore(props);
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<BrowseProjects store={store} />);
  });
  it('renders BrowseProjects', () => {
    expect(wrapper.exists()).toEqual(true);
  })
  it('renders BrowseProjects component', () => {
    wrapper = mount(<Provider store={store}><Router><BrowseProjects /></Router></Provider>);
    expect(wrapper.exists()).toEqual(true);
  });
  it('renders BrowseProjects component with 0 repo total', () => {
    props.browseResults.total = 0;
    const store = mockStore(props)
    wrapper = mount(<Provider store={store}><Router><BrowseProjects /></Router></Provider>);
    expect(wrapper.exists()).toEqual(true);
  });
  it('renders BrowseProjects component with >2 repo total', () => {
    props.browseResults.total = 2;
    const store = mockStore(props)
    wrapper = mount(<Provider store={store}><Router><BrowseProjects /></Router></Provider>);
    expect(wrapper.exists()).toEqual(true);
  });
  it('renders BrowseProjects component with >2 repo total', () => {
    props.browseResults.total = 3;
    const store = mockStore(props)
    wrapper = mount(<Provider store={store}><Router><BrowseProjects /></Router></Provider>);
    expect(wrapper.exists()).toEqual(true);
  });
  it('renders BrowseProjects component with 1 repo total', () => {
    props.browseResults.total = 1;
    const store = mockStore(props)
    wrapper = mount(<Provider store={store}><Router><BrowseProjects /></Router></Provider>);
    expect(wrapper.exists()).toEqual(true);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})