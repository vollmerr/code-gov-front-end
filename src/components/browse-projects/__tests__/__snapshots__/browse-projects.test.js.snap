// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <BrowseProjects /> component should render correctly 1`] = `
<BrowseProjects
  boxes={
    Object {
      "agencies": Array [
        Object {
          "checked": false,
          "name": "test-name",
          "value": "TV",
        },
      ],
      "languages": Array [
        Object {
          "checked": false,
          "name": "test-name",
          "value": "TV",
        },
      ],
      "licenses": Array [
        Object {
          "checked": false,
          "name": "test-name",
          "value": "TV",
        },
      ],
      "usageTypes": Array [
        Object {
          "checked": false,
          "name": "test-name",
          "value": "TV",
        },
      ],
    }
  }
  browseParams={
    Object {
      "filters": Array [],
      "page": 1,
      "size": 10,
      "sort": "data_quality",
    }
  }
  browseResults={
    Object {
      "params": Object {
        "agencies": Array [],
        "languages": Array [],
        "licenses": Array [],
        "page": 1,
        "size": 10,
        "sort": "data_quality",
        "usageTypes": Array [],
      },
      "repos": Array [
        Object {
          "repoID": "id-1",
        },
      ],
      "total": 1,
    }
  }
  filterTags={Array []}
  onFilterBoxChange={[Function]}
  onFilterTagClick={[Function]}
  onSortChange={[Function]}
  repos={
    Array [
      Object {
        "repoID": "id-1",
      },
    ]
  }
  saveFilterData={[Function]}
  selectedPage={1}
  selectedPageSize={10}
  selectedSorting="data_quality"
  sortOptions={
    Array [
      Object {
        "label": "Data Quality",
        "selected": true,
        "value": "data_quality",
      },
      Object {
        "label": "A-Z",
        "selected": false,
        "value": "a-z",
      },
      Object {
        "label": "Last Updated",
        "selected": false,
        "value": "last_update",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  total={1}
  updatePage={[Function]}
/>
`;
